pkg/asset/installconfig/aws/instancetypes.go

```diff
@@ -0,0 +1,34 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+)
+
+// InstanceType holds metadata for an instance type.
+type InstanceType struct {
+	DefaultVCpus int64
+	MemInMiB     int64
+}
+
+// instanceTypes retrieves a list of instance types for the given region.
+func instanceTypes(ctx context.Context, session *session.Session, region string) (map[string]InstanceType, error) {
+	client := ec2.New(session, aws.NewConfig().WithRegion(region))
+	resp, err := client.DescribeInstanceTypesWithContext(ctx, &ec2.DescribeInstanceTypesInput{})
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching instance types")
+	}
+
+	types := make(map[string]InstanceType, len(resp.InstanceTypes))
+	for _, item := range resp.InstanceTypes {
+		types[*item.InstanceType] = InstanceType{
+			DefaultVCpus: *item.VCpuInfo.DefaultVCpus,
```

Comment:
```suggestion
			DefaultVCpus: https://docs.aws.amazon.com/sdk-for-go/api/aws/#Int64Value(*item.VCpuInfo.DefaultVCpus),
```

Reply:
Could you explain what this does?

Reply:
```
func Int64Value
func Int64Value(v *int64) int64

Int64Value returns the value of the int64 pointer passed in or 0 if the pointer is nil. 
```

since you are referencing a pointer int64, this function makes sure it checks for nil and sets that to 0 instead of panic